modules/library/referencing/src/main/java/org/geotools/referencing/factory/gridshift/DataUtilities.java

```diff
@@ -75,27 +100,23 @@ public static File urlToFile(URL url) {
                 path3 = path2;
             }
         }
+
         return new File(path3);
     }
-    
+
     /**
-     * A replacement for File.toURI().toURL().
-     * <p>
-     * The handling of file.toURL() is broken; the handling of file.toURI().toURL() is known to be
-     * broken on a few platforms like mac. We have the urlToFile( URL ) method that is able to
-     * untangle both these problems and we use it in the geotools library.
+     * A replacement for {@link File#toURL()} and <code>File.toURI().toURL()</code>.
      * <p>
-     * However occasionally we need to pick up a file and hand it to a third party library like EMF;
-     * this method performs a couple of sanity checks which we can use to prepare a good URL
-     * reference to a file in these situtations.
+     * {@link File#toURL()} does not percent-escape characters and <code>File.toURI().toURL()</code> does not percent-escape non-ASCII characters.
+     * This method ensures that URL characters are correctly percent-escaped, and works around the reported misbehaviour of some Java implementations
+     * on Mac.
      * 
      * @param file
      * @return URL
      */
     public static URL fileToURL(File file) {
         try {
-            URL url = file.toURI().toURL();
-            String string = url.toExternalForm();
+            String string = file.toURI().toASCIIString();
```

Comment:
can you add a comment here about why this method? since this is a messed up story we may as well tell it